src/include/migraphx/op/rand_uniform.hpp

```diff
@@ -0,0 +1,128 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2015-2023 Advanced Micro Devices, Inc. All rights reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+/**
+ * Random Uniform distribution operator.  Given a shape, populate it with random values.
+ *
+ *      Inputs:   (1) the shape of the set to be populated.
+ *                (2) randomization seed.  Optional--if not given, a seed will be generated
+ *                    automatically, for nonrepeatable random results.
+ *      Attributes:  TBD
+ *
+        Output:   Same shape.
+ *
+*/
+#ifndef MIGRAPHX_GUARD_OPERATORS_MULTINOMIAL_HPP
+#define MIGRAPHX_GUARD_OPERATORS_MULTINOMIAL_HPP
+
+#include <migraphx/check_shapes.hpp>
+// #include <migraphx/argument.hpp>
+#include <migraphx/par_for.hpp>
+// #include <migraphx/reflect.hpp>
+#include <random>
+
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+namespace op {
+
+struct rand_uniform
+{
+    uint32_t sample_size = {20};
+    uint32_t seed        = {3};
+    float range_min      = 0.0f;
+    float range_max      = 1.0f;
+
+    // From Onnx RandomUniform:
+    // dtype : int (default is 1)
+    // The data type for the elements of the output tensor. If not specified, default is
+    // TensorProto::FLOAT. high : float (default is 1.0) Upper boundary of the output values. low :
+    // float (default is 0.0) Lower boundary of the output values. seed : float (Optional) Seed to
+    // the random generator, if not specified we will auto generate one. shape : list of ints
+    // (required) The shape of the output tensor.
```

Comment:
I dont think we should add the onnx documentation here as this operator will be different from the onnx spec.